src/plugins/saved_objects_management/public/management_section/objects_table/components/flyout.tsx

```diff
@@ -146,11 +146,18 @@ export class Flyout extends Component<FlyoutProps, FlyoutState> {
   import = async () => {
     const { http } = this.props;
     const { file, importMode } = this.state;
+    if (file === undefined) {
+      this.setState({
+        status: 'error',
+        error: 'missing_file',
+      });
+      return;
+    }
```

Comment:
Instead of catching this error here, WDYT about changing the error handling in the `_import` and `_resolve_import_errors` HTTP routes handlers to something more user-friendly? We have API-only consumers and they could benefit from a clearer error message.
 https://github.com/elastic/kibana/blob/dfea0fee21d93e73926092e32105d9e66fcc8c6e/src/core/server/saved_objects/routes/import.ts#L79-L86

https://github.com/elastic/kibana/blob/dfea0fee21d93e73926092e32105d9e66fcc8c6e/src/core/server/saved_objects/routes/resolve_import_errors.ts#L86-L93

Reply:
Er, actually, after re-reading the error message: `""Bad Request: [request body.file]: expected value of type [Stream] but got [string]"`

It looks like this is probably coming from the `@kbn/config-schema` validation:

https://github.com/elastic/kibana/blob/dfea0fee21d93e73926092e32105d9e66fcc8c6e/src/core/server/saved_objects/routes/import.ts#L60

https://github.com/elastic/kibana/blob/dfea0fee21d93e73926092e32105d9e66fcc8c6e/src/core/server/saved_objects/routes/resolve_import_errors.ts#L50

We could consider add custom validators there to check if `file` is empty/undefined.

OTOH, users have complained about our cryptic kbn-config validation errors for HTTP APIs before, so maybe this should be addressed separately in some Kibana-wide manner.

Reply:
I thought about adding a custom validation here but went with a custom error in the UI rather.
I've created https://github.com/elastic/kibana/issues/125280 as a follow-up for improving config validation errors.